### PR TITLE
Innovation 2022Q4 - Add a github Wiki

### PR DESCRIPTION
**Pull request by xgouchet**
### What does this PR do?

This PR adds an automated way to generate a global api reference documentation in the form of a Github Wiki

### Additional Notes

The solution uses 3 steps : 

 - Use the existing Dokka integration to generate (Github flavored) Markdown documentation 
 - Use a custom gradle plugin to generate markdown files in a simpler hierarchy that is compatible with a Github Wiki repo
 - Use a python script to update the wiki repository (triggered by a gitlab job whenever something is updated on the `master` branch)

### CODE REVIEW

**Review by 0xnm — DISMISSED** [review]
Very nice addition to the project! I've added some comments, they are non-blocking for me.

File: buildSrc/src/main/kotlin/com/datadog/gradle/plugin/wiki/GenerateWikiTask.kt [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+        val typeDir = File(packageDir, convertToDokkaTypeName(typeName))
+        val outputFile = File(outputDir, "$typeName.md")
+        pages.add(typeName)
+
+        if (typeDir.exists()) {
+            println("Combining doc from $typeCanonicalName")
+            combine(typeDir, outputFile, typeName)
+        } else {
+            System.err.println("Unable to find $typeDir")
+        }
```
[the remark this answers is not in the document]
Comment (xgouchet) [reply buildSrc/src/main/kotlin/com/datadog/gradle/plugin/wiki/GenerateWikiTask.kt]: Indeed, done

File: githubwiki.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    print("Changes added")
+
+    repo.index.commit("Update wiki")
+    print("Changes committed, head at " + repo.head.commit.hexsha)
+
+
+def git_push_changes(repo: Repo):
+    print("Pushing branch")
+    origin = repo.remote(name="origin")
+    repo.git.push("--set-upstream", "--force", origin, repo.head.ref)
```
[the remark this answers is not in the document]
Comment (xgouchet) [reply githubwiki.py]: Indeed we might not need it

File: githubwiki.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+import os
+from argparse import ArgumentParser, Namespace
+from tempfile import TemporaryDirectory
+from typing import Tuple, TextIO
+from xmlrpc.client import Boolean
+
+import requests
+from git import Repo
+
+OWNER = "xgouchet"  # "DataDog"
```
[the remark this answers is not in the document]
Comment (xgouchet) [reply githubwiki.py]: Yep, done

File: buildSrc/src/main/kotlin/com/datadog/gradle/plugin/wiki/WikiPlugin.kt [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+class WikiPlugin : Plugin<Project> {
+
+    override fun apply(target: Project) {
+
+        val task = target.tasks.register(GEN_TASK_NAME, GenerateWikiTask::class.java) {
+            this.srcDir = File(File(File(target.buildDir, "reports"), "javadoc"), target.name)
+            this.apiSurface = File(target.projectDir, "apiSurface")
+            this.outputDir = File(target.buildDir, "wiki")
+            this.projectName = target.name
+        }
```
[the remark this answers is not in the document]
Comment (xgouchet) [reply buildSrc/src/main/kotlin/com/datadog/gradle/plugin/wiki/WikiPlugin.kt]: Good idea

**Review by mariusc83 — APPROVED** [review]
Nice work !!

**Pull request closed without merging**